code/script.js

```diff
@@ -256,39 +268,78 @@ const checkQuestion = () => {
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
   if (category === 'hair' || category === 'eyes') {
-
+    if (secret[category].includes(value)) {
+      filterCharacters(true)
+    } else {
+      filterCharacters(false)
+    }
   } else if (category === 'accessories' || category === 'other') {
-
+    if (secret[category].includes(value)) {
+     filterCharacters(true)
+   } else {
+     filterCharacters(false)
+  }
   }
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
   // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  if (category === 'hair') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person has ${value} hair! Keep all people that has ${value} hair`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't have ${value} hair! Remove all people with ${value} hair`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
     }
-  } else if (category === 'other') {
+  } else if (category === 'eyes') {
     // Similar to the one above
+    if (keep) {
+      alert(
+        `Yes, the person has ${value} eyes! Keep all people that has ${value} eyes`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value);
```

Comment:
I have my "charactersInPlay=charactersInPlay.filter((person) =>....... " in a separate section, you said that didn't work for you. The difference between our code is that I decided that "keep = true", maybe that's why it works for me? But I still think it looks nice to do it your way!